### PR TITLE
Add ALCF Polaris to E3SM machines

### PR DESCRIPTION
**Pull request by amametjanov**
Add ALCF Polaris to E3SM machines.

[BFB]

### CODE REVIEW

File: cime_config/machines/config_machines.xml [hunk 1]
```diff
@@ -2955,6 +2955,101 @@
...
+  <machine MACH="polaris">
+    <DESC>ALCF Polaris 560 nodes, 2.8 GHz AMD EPYC Milan 7543P 32c CPU, 4 NVIDIA A100 GPUs</DESC>
+    <NODENAME_REGEX>polaris-*</NODENAME_REGEX>
+    <OS>Linux</OS>
+    <COMPILERS>gnu,gnugpu,nvidia,nvidiagpu</COMPILERS>
+    <MPILIBS>mpich</MPILIBS>
+    <SAVE_TIMING_DIR>/lus/grand/projects/$PROJECT</SAVE_TIMING_DIR>
+    <SAVE_TIMING_DIR_PROJECTS>ClimateEnergy_4</SAVE_TIMING_DIR_PROJECTS>
+    <CIME_OUTPUT_ROOT>/lus/swift/home/$USER/polaris/scratch</CIME_OUTPUT_ROOT>
+    <DIN_LOC_ROOT>/lus/swift/home/azamatm/polaris/inputdata</DIN_LOC_ROOT>
```
Comment (rljacob) [inline cime_config/machines/config_machines.xml:2967]: Is this a permanent location that doesn't depend on an INCITE or ALCC account?

**Comment by rljacob** [comment]
@amametjanov is this ready?

**Comment by amametjanov** [comment]
This should now be ready for review. I checked with
```
./cime/scripts/create_test --compiler gnugpu e3sm_gpucxx
./cime/scripts/create_test --compiler nvidiagpu e3sm_gpuacc
```

File: cime_config/machines/config_machines.xml [hunk 2]
```diff
@@ -3135,6 +3135,97 @@
...
+    <NODENAME_REGEX>polaris-*</NODENAME_REGEX>
+    <OS>Linux</OS>
+    <COMPILERS>gnu,gnugpu,nvidia,nvidiagpu</COMPILERS>
+    <MPILIBS>mpich</MPILIBS>
+    <PROJECT>E3SM_RRM</PROJECT>
+    <CHARGE_ACCOUNT>E3SM_RRM</CHARGE_ACCOUNT>
+    <SAVE_TIMING_DIR>/eagle/E3SM_RRM/performance_archive</SAVE_TIMING_DIR>
+    <SAVE_TIMING_DIR_PROJECTS>SCREAM_Calib,E3SM_RRM</SAVE_TIMING_DIR_PROJECTS>
+    <CIME_OUTPUT_ROOT>/eagle/E3SM_RRM/$USER/scratch</CIME_OUTPUT_ROOT>
+    <DIN_LOC_ROOT>/grand/SCREAM_Calib/inputdata</DIN_LOC_ROOT>
```
Comment (rljacob) [inline cime_config/machines/config_machines.xml:3149]: Please change this to /grand/E3SMinput   The permissions have been fixed.

File: cime_config/machines/config_machines.xml [hunk 3]
```diff
@@ -3135,6 +3135,97 @@
...
+    <DESC>ALCF Polaris 560 nodes, 2.8 GHz AMD EPYC Milan 7543P 32c CPU, 4 NVIDIA A100 GPUs</DESC>
+    <NODENAME_REGEX>polaris-*</NODENAME_REGEX>
+    <OS>Linux</OS>
+    <COMPILERS>gnu,gnugpu,nvidia,nvidiagpu</COMPILERS>
+    <MPILIBS>mpich</MPILIBS>
+    <PROJECT>E3SM_RRM</PROJECT>
+    <CHARGE_ACCOUNT>E3SM_RRM</CHARGE_ACCOUNT>
+    <SAVE_TIMING_DIR>/eagle/E3SM_RRM/performance_archive</SAVE_TIMING_DIR>
+    <SAVE_TIMING_DIR_PROJECTS>SCREAM_Calib,E3SM_RRM</SAVE_TIMING_DIR_PROJECTS>
+    <CIME_OUTPUT_ROOT>/eagle/E3SM_RRM/$USER/scratch</CIME_OUTPUT_ROOT>
```
Comment (rljacob) [inline cime_config/machines/config_machines.xml:3148]: Doe ALCF have a $PROJECT variable we could put here instead of E3SM_RRM ?

Comment (rljacob) [reply cime_config/machines/config_machines.xml:3148]: Or I guess we can use the PROJECT variable in the xml file.

File: cime_config/machines/config_machines.xml [hunk 4]
```diff
@@ -3135,6 +3135,97 @@
...
+    <DIN_LOC_ROOT>/grand/SCREAM_Calib/inputdata</DIN_LOC_ROOT>
+    <DIN_LOC_ROOT_CLMFORC>/grand/SCREAM_Calib/inputdata/atm/datm7</DIN_LOC_ROOT_CLMFORC>
```
Comment (rljacob) [inline cime_config/machines/config_machines.xml:3150]: Fix this as well.

File: cime_config/machines/config_machines.xml [hunk 5]
```diff
@@ -3135,6 +3135,97 @@
...
+    <DOUT_S_ROOT>$CIME_OUTPUT_ROOT/archive/$CASE</DOUT_S_ROOT>
+    <BASELINE_ROOT>/eagle/E3SM_RRM/baselines/$COMPILER</BASELINE_ROOT>
```
Comment (rljacob) [inline cime_config/machines/config_machines.xml:3152]: We need a better place for this too.

**Review by rljacob — CHANGES_REQUESTED** [review]
Need to change some paths.

**Review by rljacob — APPROVED** [review]

**Pull request merged**